cmd/uhc/cluster/status/status.go

```diff
@@ -0,0 +1,142 @@
+	parameter []string
+	header    []string
+}
+
+var Cmd = &cobra.Command{
+	Use:   "status CLUSTERNAME ",
+	Short: "status of a cluster",
+	Long:  "Get the status of a cluster identified by its cluster name",
+	Run:   run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+	flags.BoolVar(
+		&args.debug,
+		"debug",
+		false,
+		"Enable debug mode.",
+	)
+	flags.StringArrayVar(
+		&args.parameter,
+		"parameter",
+		nil,
+		"Query parameters to add to the request. The value must be the name of the "+
+			"parameter, followed by an optional equals sign and then the value "+
+			"of the parameter. Can be used multiple times to specify multiple "+
+			"parameters or multiple values for the same parameter.",
+	)
+	flags.StringArrayVar(
+		&args.header,
+		"header",
+		nil,
+		"Headers to add to the request. The value must be the name of the header "+
+			"followed by an optional equals sign and then the value of the "+
+			"header. Can be used multiple times to specify multiple headers "+
+			"or multiple values for the same header.",
+	)
+}
+func run(cmd *cobra.Command, argv []string) {
+
+	if len(argv) != 1 {
+		fmt.Fprintf(os.Stderr, "Expected exactly one cluster\n")
+		os.Exit(1)
+	}
+
+	clusterName := "name='" + argv[0] + "'"
+
+	// Load the configuration file:
+	cfg, err := config.Load()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't load config file: %v\n", err)
+		os.Exit(1)
+	}
+	if cfg == nil {
+		fmt.Fprintf(os.Stderr, "Not logged in, run the 'login' command\n")
+		os.Exit(1)
+	}
+
+	// Check that the configuration has credentials or tokens that haven't have expired:
+	armed, err := config.Armed(cfg)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't check if tokens have expired: %v\n", err)
+		os.Exit(1)
+	}
+	if !armed {
+		fmt.Fprintf(os.Stderr, "Tokens have expired, run the 'login' command\n")
+		os.Exit(1)
+	}
+
+	// Create the connection:
+	logger, err := util.NewLogger(args.debug)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't create logger: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Create the connection, and remember to close it:
+	connection, err := client.NewConnectionBuilder().
+		Logger(logger).
+		TokenURL(cfg.TokenURL).
+		Client(cfg.ClientID, cfg.ClientSecret).
+		Scopes(cfg.Scopes...).
+		URL(cfg.URL).
+		User(cfg.User, cfg.Password).
+		Tokens(cfg.AccessToken, cfg.RefreshToken).
+		Insecure(cfg.Insecure).
+		Build()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't create connection: %v\n", err)
+		os.Exit(1)
+	}
+	defer connection.Close()
+
+	// Get the client for the resource that manages the collection of clusters:
+	collection := connection.ClustersMgmt().V1().Clusters()
+
+	// Retrieve the collection of clusters:
+	response, err := collection.List().
+		Search(clusterName).
+		Send()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Can't retrieve clusters: %s", err)
+		os.Exit(1)
+	}
+
+	// Print the result:
+	response.Items().Each(func(cluster *v1.Cluster) bool {
+		fmt.Printf("%s\n", cluster.State())
+		return true
+	})
```

Comment:
Is this assuming that there is only one cluster with a particular name? There is no such restriction in the server side, there may be thousands of clusters with the same name, so you will need to use paging, and display them in a way that is compatible with multiple clusters.

Reply:
Is a cluster unique by ID then? The goal is to get status of a specific cluster.

Reply:
Yes, the cluster `id` is unique. If you want to get the details of a cluster, and you know the `id` then you can get it as explained in [this example](https://github.com/openshift-online/uhc-sdk-go/blob/master/examples/get_cluster.go).

Reply:
:+1: I will use that